src/Routes.php

```diff
@@ -55,6 +69,37 @@ public function endpoint_callback( \WP_REST_Request $request ) {
 
 		wp_reset_postdata();
 
+		// Create routes for the blog page and single posts if active.
```

Comment:
Can you spit this logic in different methos at this point `endpoint_callback` is too complex.